backend/aci/common/schemas/subscription.py

```
@@ -27,13 +28,32 @@ class SubscriptionUpdate(BaseModel):
+        """
+        Get the current period start date.
+
+        - Free plans: 1st of current month at 00:00:00 UTC (calculated dynamically)
+        - Paid plans: Use the stored current_period_start from Stripe
+        """
+        if self.plan.name == "free" or self._stripe_current_period_start is None:
+            # For free plans, always calculate the current month's start
+            return datetime.now(UTC).replace(day=1, hour=0, minute=0, second=0, microsecond=0)
+        else:
+            # For paid plans, use the actual Stripe billing period
+            return self._stripe_current_period_start.replace(tzinfo=UTC)
```

Pylint (3.3.7)

[refactor] Unnecessary "else" after "return", remove the "else" and de-indent the code inside it

(R1705)